tests/rkt_auth_test.go

```diff
@@ -0,0 +1,289 @@
+}
+
+func oauthNoConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	runRkt401(t, server.URL, "oauth1")
+}
+
+func oauthCustomConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeCustomConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-custom-config")
+}
+
+func oauthVendorConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-vendor-config")
+}
+
+func TestAuthOverride(t *testing.T) {
+	skipUnsafe(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	validVendorInvalidCustom(t, server)
+	invalidVendorValidCustom(t, server)
+}
+
+func validVendorInvalidCustom(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-vvic-1")
+	writeCustomConfig(t, "test.json", getInvalidOAuthConfig(server.Conf))
+	runRkt401(t, server.URL, "oauth-vvic-2")
+}
+
+func invalidVendorValidCustom(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", getInvalidOAuthConfig(server.Conf))
+	runRkt401(t, server.URL, "oauth-ivvc-1")
+	writeCustomConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-ivvc-2")
+}
+
+func TestAuthIgnore(t *testing.T) {
+	skipUnsafe(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	bogusFiles(t, server)
+	subdirectories(t, server)
+}
+
+func bogusFiles(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "README", "This is vendor config")
+	writeCustomConfig(t, "README", "This is custom config")
+	writeVendorConfig(t, "test.notjson", server.Conf)
+	writeCustomConfig(t, "test.notjson", server.Conf)
+	runRkt401(t, server.URL, "oauth-bogus-files")
+}
+
+func subdirectories(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	customSubdir := filepath.Join(config.DefaultCustomPath, "subdir")
+	vendorSubdir := filepath.Join(config.DefaultVendorPath, "subdir")
+	writeConfig(t, customSubdir, "test.json", server.Conf)
+	writeConfig(t, vendorSubdir, "test.json", server.Conf)
+	runRkt401(t, server.URL, "oauth-subdirectories")
+}
+
+// TODO (krnowak): Remove this when we will be able to specify
+// different custom and vendor configuration directories.
+const unsafeEnvVar = "DELETE_DATA_AND_CONFIG"
+
+func skipUnsafe(t *testing.T) {
+	if os.Getenv(unsafeEnvVar) != "1" {
+		t.Skipf("%s envvar is not specified or has value different than 1, skipping the test", unsafeEnvVar)
+	}
+}
+
+func removeDataDir(t *testing.T) {
+	if err := os.RemoveAll("/var/lib/rkt"); err != nil {
+		t.Fatalf("Failed to remove /var/lib/rkt: %v", err)
+	}
+}
+
+func runServer(t *testing.T, auth taas.Type) *taas.Server {
+	goTool := getAbs(t, "go")
+	acTool := getAbs(t, "../bin/actool")
+	server, err := taas.NewServerWithPaths(auth, 20, acTool, goTool)
```

Comment:
Could taas.NewServerWithPaths() be changed to accept non-absolute paths as well? This would be less code here. And taas already code to do the equivalent of getAbs() in getTool().